src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleMetadata.cs

```diff
@@ -222,7 +222,7 @@ public static partial class JsonSerializer
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static bool ReadAheadMetataDataAndSetState(ref Utf8JsonReader reader, ref ReadStack state, StackFrameObjectState nextState)
+        private static bool ReadAheadMetadataAndSetState(ref Utf8JsonReader reader, ref ReadStack state, StackFrameObjectState nextState)
         {
```

Comment:
Can you add a comment explaining that we always move the state because if Read return false, the token will be read by the re-entry mechanism on `SingleValueReadWithReadAhead`?

Reply:
Will do in a follow up.